cloudflare/resource_cloudflare_record.go

```diff
@@ -128,16 +177,18 @@ func resourceCloudFlareRecordCreate(d *schema.ResourceData, meta interface{}) er
 
 func resourceCloudFlareRecordRead(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*cloudflare.API)
-	domain := d.Get("domain").(string)
+	zoneID := d.Get("zone_id").(string)
 
-	zoneId, err := client.ZoneIDByName(domain)
+	record, err := client.DNSRecord(zoneID, d.Id())
 	if err != nil {
-		return fmt.Errorf("Error finding zone %q: %s", domain, err)
-	}
-
-	record, err := client.DNSRecord(zoneId, d.Id())
-	if err != nil {
-		return err
+		if strings.Contains(err.Error(), "Invalid dns record identifier") ||
+			strings.Contains(err.Error(), "HTTP status 404") {
+			log.Printf("[WARN] Removing record from state because it's not found in API")
+			d.SetId("")
+			return nil
+		} else {
```

Comment:
nit: no need for else here because of the return.

```go
if err!= nil {
  if stringsContains .... {
    /// 
    d.SetId("")
    return nil
  }
  
  return err
}
```

Reply:
Applied in https://github.com/terraform-providers/terraform-provider-cloudflare/commit/f01532c4a37f894b2926b54bed9e5d21d19255cd